movement/watch_faces/settings/set_time_face.c

```diff
@@ -72,22 +52,79 @@ bool set_time_face_loop(movement_event_t event, movement_settings_t *settings, v
                 case 4: // month
                     date_time.unit.month = (date_time.unit.month % 12) + 1;
                     break;
-                case 5: // day
-                    date_time.unit.day = date_time.unit.day + 1;
-                    // can't set to the 29th on a leap year. if it's february 29, set to 11:59 on the 28th.
-                    // and it should roll over.
-                    if (date_time.unit.day > days_in_month[date_time.unit.month - 1]) {
-                        date_time.unit.day = 1;
+        case 5: { // day
+            uint32_t tmp_day = date_time.unit.day;   // use a temporary variable to avoid messing up the months
+            tmp_day = tmp_day + 1;
+            // handle February 29th on a leap year
+            if (((tmp_day > days_in_month[date_time.unit.month - 1]) && (date_time.unit.month != 2 || (date_time.unit.year % 4) != 0))
```

Comment:
I think there's an `is_leap` in watch_utility that will leap year calculation fully

Reply:
I looked and could not find such a function. Besides, the condition for checking for leap years is rather trivial, so I am not sure using a function would be necessary here.